<launch>
  <arg name="gui" default="true"/>

  <node name="static_tf_publisher"
        pkg="tf" type="static_transform_publisher"
        args="0 0 3 0 0 0 /map /dummy_frame 10"/>

  <node name="sample_manager"
        pkg="nodelet" type="nodelet"
        args="manager"/>

  <node name="static_polygon_array_publisher"
        pkg="nodelet" type="nodelet"
        args="load jsk_pcl_utils/StaticPolygonArrayPublisher sample_manager">
    <rosparam>
      use_periodic: true
      periodic_rate: 1.0
      frame_ids:
        - map
        - map
        - map
        - map
      polygon_array:
        - [[-2.0, 2.0, 2.0], [-2.0, 0.0, 2.0], [0.0, 0.0, 2.0], [0.0, 2.0, 2.0]]
        - [[2.0, 2.0, 2.0], [2.0, 0.0, 2.0], [0.0, 0.0, 2.0], [0.0, 2.0, 2.0]]
        - [[-2.0, -2.0, 2.0], [-2.0, 0.0, 2.0], [0.0, 0.0, 2.0], [0.0, -2.0, 2.0]]
        - [[2.0, -2.0, 2.0], [2.0, 0.0, 2.0], [0.0, 0.0, 2.0], [0.0, -2.0, 2.0]]
    </rosparam>
  </node>

  <node name="polygon_flipper"
        pkg="nodelet" type="nodelet"
        args="load jsk_pcl_utils/PolygonFlipper sample_manager">
    <remap from="~input/polygons" to="static_polygon_array_publisher/output_polygons"/>
    <remap from="~input/coefficients" to="static_polygon_array_publisher/output_coefficients"/>
    <rosparam>
      sensor_frame: dummy_frame
      use_indices: false
    </rosparam>
  </node>

  <group if="$(arg gui)">
    <node name="rviz"
          pkg="rviz" type="rviz"
          args="-d $(find jsk_pcl_ros_utils)/sample/config/sample_polygon_flipper.rviz"/>
  </group>
</launch>
